techniques/applications/systemUpdate/1.0/systemUpdate.st

```diff
@@ -83,48 +85,41 @@ bundle agent rudder_system_update {
       "enable_reporting"   usebundle => enable_reporting;
 
       # Pre-Update
-    pass1::
-      "Setup variable context" usebundle => _method_reporting_context_v4("Setup variable context", "Setup variable context", "Setup variable context");
-      "Unsupported OS"         usebundle => report_if_condition("variable_reporting", "use_yum_provider|use_apt_provider|use_zypper_provider");
-
+    pass2::
+      "Setup variable context" usebundle => _method_reporting_context_v4("OS Support", "", "system-update");
+      "Setup variable context" usebundle => _rudder_report_state("system-update", "Unsupported OS for system update", "result_error"),
+                                  unless => or("use_yum_provider", "use_apt_provider", "use_zypper_provider");
     pass3::
       # Update
-      "Update the system"     usebundle => _method_reporting_context_v4("Update the system", "Update the system", "Update the system");
+      "Update the system"     usebundle => _method_reporting_context_v4("Update the system", "", "system-update");
       "Update the system"     usebundle => command_execution_once("${update_command}", "0", "any", "${campaign_name}"),
                                      if => and("update", "must_run_system_update", "!flag_set");
 
-      "NA report by conf"     usebundle => log_na_rudder( "An update is needed but the update step is disabled by the directive configuration",
-                                                          "Update the system",
-                                                          "rudder_system_update_reboot",
-                                                          @{args}
-                                                        ),
-                                    if  => not("update");
-      "NA report by schedule" usebundle => log_na_rudder( "${na_message}",
-                                                          "Update the system",
-                                                          "rudder_system_update_reboot",
-                                                          @{args}
-                                                        ),
-                                     if => or( and("update", not("must_run_system_update")),
+      "NA update" usebundle => _rudder_report_state("system-update", "${na_message}", "result_na"),
+                                     if => or( not("update"),
+                                               and("update", not("must_run_system_update")),
                                                and("update",
                                                    "must_run_system_update",
                                                    "flag_set",
                                                    not(canonify("command_execution_once_${update_command}_reached")))
```

Comment:
I think this one is redundant, as if flag_set, the command will never be executed